ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,284 @@
+            onlineMap: onlineMap
+
+            onErrorChanged: console.log(error.message, error.additionalMessage);
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                offlineMapTask.preplannedMapAreas();
+                path = outputMapPackage;
+                fileFolder.url = path;
+                let worked = fileFolder.makeFolder();
+                if (!worked)
+                    console.log("Make Folder Failed");
+            }
+
+            onPreplannedMapAreasStatusChanged: {
+                if (preplannedMapAreasStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                preplannedCombo.model = offlineMapTask.preplannedMapAreaList;
+                busy.visible = false;
+
+                for (let i = 0; i < offlineMapTask.preplannedMapAreaList.count; i++) {
+                    offlineMapTask.preplannedMapAreaList.get(i).loadStatusChanged.connect(function () {
+                        if (offlineMapTask.preplannedMapAreaList.get(i).loadStatus !== Enums.LoadStatusLoaded)
+                            return;
+
+                        if (i === 0)
+                            checkIfFileExists(i);
+
+                        var graphic = ArcGISRuntimeEnvironment.createObject("Graphic", { geometry: offlineMapTask.preplannedMapAreaList.get(i).areaOfInterest });
+
+                        graphicsOverlay.graphics.append(graphic);
+                    });
+                    offlineMapTask.preplannedMapAreaList.get(i).load();
+                }
+            }
+
+            onCreateDefaultDownloadPreplannedOfflineMapParametersStatusChanged: {
+                if (createDefaultDownloadPreplannedOfflineMapParametersStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                createDefaultDownloadPreplannedOfflineMapParametersResult.updateMode = Enums.PreplannedUpdateModeNoUpdates;
+                let result = createDefaultDownloadPreplannedOfflineMapParametersResult;
+
+                path = outputMapPackage +"/" + result.preplannedMapArea.portalItem.title + ".mmpk";
+                fileFolder.url = path;
+
+                if (fileFolder.exists) {
+                    var mmpk = ArcGISRuntimeEnvironment.createObject("MobileMapPackage", { path: path });
+                    mmpk.loadStatusChanged.connect(function () {
+                        if (loadStatus !== Enums.LoadStatusLoaded )
+                            return;
+
+                        if (mmpk.maps.length < 1)
+                            return;
+
+                        mapView.map = mmpk.maps[0];
+                        busy.visible = false;
+                    });
+                    mmpk.load();
+                    return;
+                }
+
+                var job = offlineMapTask.downloadPreplannedOfflineMapWithParameters(createDefaultDownloadPreplannedOfflineMapParametersResult, path);
```

Comment:
you might want to see about storing this variable at global scope. It's possible the function returns and the gc runs before the jobStatusChanged.connect function executes